Add app name to include generation if running in Edge Code

joelrbrandt opened the pull request.

joelrbrandt committed:
add app name to include if running in Edge Code

gruehle commented on webfont.js:
```diff
@@ -349,6 +352,21 @@ define(function (require, exports, module) {
+                appNameInclude = "<script>var __adobewebfontsappname__ = \"code\"</script>\n";
+            }
+        } catch (err) {
+            console.log("[edge-web-font extension] failed to compute app identifier: " + err);
```
Should we even bother with the console.log here? It's no big deal if the code fails, and the message doesn't have any actionable content anyway.

joelrbrandt replied:
@gruehle I'm all for keeping the console clean, so I'll take this line out.

However, because I love being pedantic: I thought about this when I wrote it. If the user hits the "catch" case, we are catching an _actual_ bug that should be fixed. It seems to me that it's good programming practice to make sure that actual bugs have an outwardly-visible symptom. (Though it's not clear that this console.log is the right symptom.)

gruehle commented:
Done with initial review. Just one minor comment.

joelrbrandt committed:
remove console.log

joelrbrandt commented:
@gruehle Thanks for the quick review! I removed the console.log.

I also added this pedantic thought, which is hidden because I deleted the line in question:

> I'm all for keeping the console clean, so I'll take this line out.
> 
> However, because I love being pedantic: I thought about this when I wrote it. If the user hits the "catch" case, we are catching an actual bug that should be fixed. It seems to me that it's good programming practice to make sure that actual bugs have an outwardly-visible symptom. (Though it's not clear that this console.log is the right symptom.)

gruehle commented:
I guess I would argue that an exception here is _always_ indicative of a bug. The required functionality is having the appname script included when run from Edge Code (or any other brackets derivative whose name ends with " code"). That result _is_ outwardly visible--you either have the appname script or you don't. 

Merging.
